pkg/pkg.bzl

```diff
@@ -80,27 +80,26 @@ def _pkg_tar_impl(ctx):
     # Find a list of path remappings to apply.
     remap_paths = ctx.attr.remap_paths
 
+    # Start building the arguments.
+    args = ctx.actions.args()
+    args.add("--root_directory", ctx.attr.package_base)
+    args.add("--output", output_file.path)
+    args.add("--mode", ctx.attr.mode)
+    args.add("--owner", ctx.attr.owner)
+    args.add("--owner_name", ctx.attr.ownername)
+
     # Package dir can be specified by a file or inlined.
     if ctx.attr.package_dir_file:
         if ctx.attr.package_dir:
             fail("Both package_dir and package_dir_file attributes were specified")
-        package_dir_arg = "--directory=@" + ctx.file.package_dir_file.path
+        args.add("--directory", "@" + ctx.file.package_dir_file.path)
```

Comment:
Do we really need to get the `path` here?  Would something like this work?

```starlark
args.add("--directory", ctx.file.package_dir_file, format = "@%s")
```

Reply:
That will give us the same thing. 
The rest of the code is explicit using `<f>.path` at the boundary between rule and the action. I think that is clearer than relying on the implicit conversion of file to file.path done by `%s`. 
Then there is the question of which is more efficient. The format solution might be slightly better, but this is only one arg for the action. Overall there is only the copy of this one path to save.